rpc/core/blocks.go

```diff
@@ -357,17 +366,63 @@ func BlockResults(heightPtr *int64) (*ctypes.ResultBlockResults, error) {
 		return nil, err
 	}
 
-	// load the results
-	results, err := sm.LoadABCIResponses(stateDB, height)
-	if err != nil {
-		return nil, err
+	// for last height, use state DB
```

Comment:
Why bother with the special case? Can't we just get everything through the indexer?

Reply:
But if the indexer is disabled, you won't get anything. Now, even though it's off, you can still view the results for the last height.